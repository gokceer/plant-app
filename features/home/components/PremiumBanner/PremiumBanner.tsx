import { GradientText } from "@/components";
import React from "react";
import { Image, Pressable, StyleSheet, View } from "react-native";

const PremiumBanner = ({ onPress }: { onPress?: () => void }) => {
  return (
    <Pressable
      style={styles.premiumBanner}
      onPress={() => onPress?.()}
      testID="premiumBannerButton"
    >
      <Image
        source={require("../../../../assets/icons/envelopeIcon.png")}
        style={styles.envelopeIcon}
      />
      <View style={styles.premiumTextContainer}>
        <GradientText
          text="FREE Premium Available"
          style={{ fontSize: 16, fontFamily: "RubikBold" }}
          colors={["#E5C990", "#E4B046"]}
        />
        <GradientText
          text="Tap to upgrade your account!"
          style={styles.premiumSubtitle}
          colors={["#FFDE9C", "#F5C25B"]}
        />
      </View>
      <Image
        source={require("../../../../assets/icons/arrowIcon.png")}
        style={styles.arrowIcon}
      />
    </Pressable>
  );
};

export default PremiumBanner;

const styles = StyleSheet.create({
  premiumBanner: {
    flexDirection: "row",
    alignItems: "center",
    backgroundColor: "#24201A",
    borderRadius: 12,
    padding: 16,
    margin: 24,
  },
  envelopeIcon: {
    width: 48,
    height: 40,
    textAlign: "center",
    alignSelf: "center",
  },
  arrowIcon: {
    width: 24,
    height: 24,
  },
  premiumTextContainer: {
    flex: 1,
    marginLeft: 12,
    justifyContent: "center",
  },
  premiumTitle: {
    color: "#FFD700",
    fontWeight: "bold",
    fontSize: 16,
  },
  premiumSubtitle: {
    color: "#fff",
    fontSize: 13,
    fontFamily: "RubikRegular",
  },
});
