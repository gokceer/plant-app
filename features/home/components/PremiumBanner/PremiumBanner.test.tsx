import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import PremiumBanner from "./PremiumBanner";

jest.mock("@/components/GradientText/GradientText", () => ({
  __esModule: true,
  default: jest.fn(({ text }) => <>{text}</>),
}));

describe("PremiumBanner Component", () => {
  const mockOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly and matches snapshot", () => {
    const { toJSON } = render(<PremiumBanner onPress={mockOnPress} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("calls onPress when pressed", () => {
    const { getByTestId } = render(<PremiumBanner onPress={mockOnPress} />);
    fireEvent.press(getByTestId("premiumBannerButton"));
    expect(mockOnPress).toHaveBeenCalled();
  });
});
