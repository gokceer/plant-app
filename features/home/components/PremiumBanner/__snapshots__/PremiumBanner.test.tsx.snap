// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PremiumBanner Component renders correctly and matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#24201A",
      "borderRadius": 12,
      "flexDirection": "row",
      "margin": 24,
      "padding": 16,
    }
  }
  testID="premiumBannerButton"
>
  <Image
    source={
      {
        "testUri": "../../../assets/icons/envelopeIcon.png",
      }
    }
    style={
      {
        "alignSelf": "center",
        "height": 40,
        "textAlign": "center",
        "width": 48,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 12,
      }
    }
  >
    FREE Premium Available
    Tap to upgrade your account!
  </View>
  <Image
    source={
      {
        "testUri": "../../../assets/icons/arrowIcon.png",
      }
    }
    style={
      {
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;
